### PR TITLE
Adding features arg to clustering CV, removing test

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Addressing comments in #1584, which was merged before comments could be addressed.

### CODE REVIEW

File: src/Microsoft.ML.Data/TrainContext.cs [hunk 1]
```diff
@@ -331,7 +331,7 @@ internal ClusteringTrainers(ClusteringContext ctx)
                 Host.CheckNonEmpty(features, nameof(features), "The features column name should be non-empty, if provided, if you want to calculate the Dbi metric.");
 
             if (label != null)
-                Host.CheckNonEmpty(label, nameof(label), "The features column name should be non-empty, if provided, if you want to calculate the Nmi metric.");
+                Host.CheckNonEmpty(label, nameof(label), "The label column name should be non-empty, if provided, if you want to calculate the Nmi metric.");
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/TrainContext.cs]: >The label column name should be non-empty, if provided, if you want to calculate the Nmi metric [](start = 58, length = 95)

"To calculate the NMI metric,  provide a value for the the optional label column".

File: src/Microsoft.ML.Data/TrainContext.cs [hunk 2]
```diff
@@ -346,16 +346,17 @@ internal ClusteringTrainers(ClusteringContext ctx)
         /// <param name="estimator">The estimator to fit.</param>
         /// <param name="numFolds">Number of cross-validation folds.</param>
         /// <param name="labelColumn">Optional label column for evaluation (clustering tasks may not always have a label).</param>
+        /// <param name="featuresColumn">OPtional features column for evaluation (needed for calculating Davies-Bouldin Index)</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/TrainContext.cs]: >OPtional [](start = 41, length = 8)

Optional

File: src/Microsoft.ML.Data/TrainContext.cs [hunk 3]
```diff
@@ -346,16 +346,17 @@ internal ClusteringTrainers(ClusteringContext ctx)
...
         /// <param name="stratificationColumn">Optional stratification column.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/TrainContext.cs]: >Optional stratification column. [](start = 47, length = 31)

maybe a sentence on what this is used for.

File: src/Microsoft.ML.Data/TrainContext.cs [hunk 4]
```diff
@@ -346,16 +346,17 @@ internal ClusteringTrainers(ClusteringContext ctx)
...
         /// <remarks>If two examples share the same value of the <paramref name="stratificationColumn"/> (if provided),
         /// they are guaranteed to appear in the same subset (train or test). Use this to make sure there is no label leakage from
         /// train to the test set.</remarks>
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/TrainContext.cs]: makes more sense to me to have this comment on the strat col param.

Comment (najeeb-kazmi) [reply src/Microsoft.ML.Data/TrainContext.cs]: Why? Stratification column is an argument to `CrossValidateTrain`, not `ClusteringEvaluator.Evaluate`. It does not affect the calculation of BDI.

**Pull request merged**